### PR TITLE
Light replacer use in hand to replace lamps in area

### PR DESCRIPTION
**Pull request by Saicchi**
## About the Pull Request

Lets the light replacer replace all broken adjacent lights after a 1 second do_after

## Why It's Good For The Game

Clicking the tiny pixels of a broken light fixture is hard.

## Changelog

:cl:
add: light replacer replace lights in area
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/lightreplacer.dm [hunk 1]
```diff
@@ -116,15 +116,16 @@
...
 /obj/item/device/lightreplacer/attack_self(mob/user)
-	/* // This would probably be a bit OP. If you want it though, uncomment the code.
-	if(isrobot(user))
-		var/mob/living/silicon/robot/R = user
-		if(R.emagged)
-			src.Emag()
-			to_chat(usr, "You shortcircuit the [src].")
-			return
-	*/
-	to_chat(usr, "It has [uses] lights remaining.")
```
Comment (checkraisefold) [inline code/game/objects/items/devices/lightreplacer.dm:127]: ?

Comment (Saicchi) [reply code/game/objects/items/devices/lightreplacer.dm:127]: The user can see the remaining lights by examining the tool.

Comment (checkraisefold) [reply code/game/objects/items/devices/lightreplacer.dm:127]: > The user can see the remaining lights by examining the tool.

oh this is fine then

File: code/game/objects/items/devices/lightreplacer.dm [hunk 2]
```diff
@@ -116,15 +116,16 @@
...
+	if(!isliving(user) || !do_after(user, 1 SECONDS, src))
+		return
+
+	for(var/obj/machinery/light/O in range(1,user))
+		if (!CanUse(user))
```
Comment (TheDarkElites) [inline code/game/objects/items/devices/lightreplacer.dm]: This should just be an early return outside of the loop instead of a break statement inside

**Review by TheDarkElites — CHANGES_REQUESTED** [review]

File: code/game/objects/items/devices/lightreplacer.dm (hunk 2 above)
Comment (Saicchi) [reply code/game/objects/items/devices/lightreplacer.dm]: Changed it.

Comment (TheDarkElites) [reply code/game/objects/items/devices/lightreplacer.dm]: When I say early return I mean having the return statement **outside of the for loop**

Comment (Saicchi) [reply code/game/objects/items/devices/lightreplacer.dm]: Rewrote it a bit to move the check to outside the loop and give a small delay when replacing each light. Does it look better now?

**Review by TheDarkElites — APPROVED** [review]

**Pull request merged**